js/classes/entities/midground/enemies/ground_enemy_base.js

```diff
@@ -0,0 +1,199 @@
+        this.getFacing = enemy.getFacing; 
+        this.getCenter = enemy.getCenter;
+    }
+
+    /** 
+     * Aggressive enemy stance. Aggressive enemies will pursue Chad any time he
+     * is within their maximum reaction distance. 
+     */
+    static get AGGRESSIVE_STANCE() {
+        return "aggressive";
+    }
+
+    /** 
+     * Defensive enemy stance. Defensive enemies will pursue/attack Chad only when
+     * provoked.
+     */
+    static get DEFENSIVE_STANCE() {
+        return "defensive";
+    }
+
+    /** Passive enemy stance. Passive enemies will run from Chad when attacked. */
+    static get PASSIVE_STANCE() {
+        return "passive";
+    }
+
+    /** 
+     * Avoid enemy stance. Enemies with the avoid stance will run from Chad whenever
+     * he is within their maximum reaction distance.
+     */
+    static get AVOID_STANCE() {
+        return "avoid";
+    }
+
+    /** The initial velocity, in pixels/s of an enemy's jump. */
+    get jumpVelocity() {
+        return 450;
+    }
+    
+    /**
+     * Make the enemy react to taking damage.
+     */
+    handleDamage() {
+        switch (this.stance) {
+            case GroundEnemyBase.AGGRESSIVE_STANCE:
+            case GroundEnemyBase.DEFENSIVE_STANCE:
+                this.enemy.state = "pursue";
+                break;
+            case GroundEnemyBase.PASSIVE_STANCE:
+            case GroundEnemyBase.AVOID_STANCE:
+                this.flee();
+        }
+    }
+
+    /**
+     * Get the direction in which the enemy is currently facing.
+     * 
+     * @returns {number} -1 for left and 1 for right
+     */
+    getDirection() {
+        return this.targetX - this.enemy.getCenter().x > 0 ? 1 : -1;
+    }
+
+    /**
+     * Set the target x-coordinate of the enemy.
+     * 
+     * @param {number} targetX the x-coordinate the enemy should move towards
+     */
+    setTargetX(targetX) {
+        this.targetX = targetX;
+    };
+
+    /**
+     * Makes the enemy run away from Chad. Also changes its stance to avoid.
+     */
+    flee() {
+        this.setTargetX(this.enemy.pos.x + Math.sign(this.enemy.pos.x - CHAD.pos.x) * this.reactionDistance);
+        this.stance = EnemyBase.AVOID_STANCE;
+        this.enemy.state = "flee";
+    }
+
+    /**
+     * Makes the enemy pursue Chad. Also changes its stance to aggressive.
+     */
+    pursue() {
+        this.stance = EnemyBase.AGGRESSIVE_STANCE;
+        this.enemy.state = "pursue";
+    }
+
+    /**
+     * Calculates the distance between Chad and the enemy, comparing their bottom left corners
+     * in order to avoid having to account for character/enemy height.
+     * 
+     * @returns {number} the distance between the bottom left corners of Chad and the enemy
+     */
+    chadDistance() {
+        return Vector.distance(Vector.add(CHAD.getCenter(), new Vector(0, CHAD.scaledSize.y / 2)),
+            Vector.add(this.enemy.getCenter(), new Vector(0, this.enemy.scaledSize.y / 2)));
```

Comment:
Why is our second vector such as: `new Vector(0, CHAD.scaledSize.y / 2)` and `new Vector(0, this.enemy.scaledSize.y / 2)` using 0 for the first value in the vector and a this calculation in the second? Is it not just the center that we would want?

Reply:
This method finds distance by comparing the bottom-center point of each entity. Doing that allows the distance to be unaffected by the height of the enemy.

Reply:
Ah good idea.